R/methods.R

```diff
@@ -981,9 +984,9 @@ plot.vsel <- function(
       thres_tab_ref$thres[is_elpd_mlpd_ref] <-
```

Comment:
I can't comment on the line before this one, so I'll comment here:

For elpd and mlpd, don't we need to take the new `_diff` variants into account (similarly to gmpd where `gmpd_ratio` was taken into account below)?

Reply:
This is related to the change in deltas=TRUE

Reply:
Ah, I now see that line https://github.com/stan-dev/projpred/blob/c23fa783b1dc044cbe4e52e25682e530192f6fde/R/methods.R#L988 has `"gmpd ratio"` and not `"gmpd_ratio"` (which is used in lines https://github.com/stan-dev/projpred/blob/c23fa783b1dc044cbe4e52e25682e530192f6fde/R/methods.R#L966-L967). So I guess that `"gmpd ratio"` without underscore has no effect.

More importantly, since we have several (two, I think) `merge(thres_tab_basic, [...], by = "statistic")` calls, the names of the statistics in `thres_tab_basic` would also need to be adapted analogously to lines https://github.com/stan-dev/projpred/blob/c23fa783b1dc044cbe4e52e25682e530192f6fde/R/methods.R#L962-L967. The same would apply to `stats_bs`, I think. For example, at the current state of this PR (commit c23fa783b1dc044cbe4e52e25682e530192f6fde),
```r
data("df_gaussian", package = "projpred")
dat <- data.frame(y = df_gaussian$y, df_gaussian$x)
rfit <- rstanarm::stan_glm(y ~ X1 + X2 + X3 + X4 + X5,
                           data = dat,
                           chains = 1,
                           iter = 500,
                           seed = 1140350788,
                           refresh = 0)

devtools::load_all(".")

vs <- varsel(rfit,
             nclusters = 3,
             nclusters_pred = 5,
             nterms_max = 2,
             seed = 46782345)

plot(vs, thres_elpd = -4, deltas = TRUE)
plot(vs, thres_elpd = -4, deltas = TRUE, baseline = "best")
```
produces the following two plots:

![Rplot1_new](https://github.com/stan-dev/projpred/assets/55132727/8a93041f-419d-4fa6-92b2-a461246ced5f)
![Rplot2_new](https://github.com/stan-dev/projpred/assets/55132727/f828be3b-e5fd-4ffb-99eb-e48317f60d0f)

whereas on `master` (current commit: 28e011789ea2b5ad2743c7821bc88b2ca2c76a77), we get

![Rplot1_old](https://github.com/stan-dev/projpred/assets/55132727/3fa551fe-6c09-41b6-90fc-96f87930a755)
![Rplot2_old](https://github.com/stan-dev/projpred/assets/55132727/79eb0ec6-e8db-4e1b-825d-3f4f926cd8e1)

But before now adapting `thres_tab_basic` and `stats_bs` (and possibly other places within `plot.vsel()`), I think we should clarify whether this is really necessary. Reason 1: The `**pd` statistics should not be affected by the new `deltas = TRUE` behavior. Reason 2: If `deltas = TRUE`, the *baseline model's* (here, it has to be kept in mind that the baseline model might be a different model than the reference model) ELPD(-diff) and MLPD(-diff) should be zero, so adding it to the threshold does not change the threshold (for example, in lines https://github.com/stan-dev/projpred/blob/28e011789ea2b5ad2743c7821bc88b2ca2c76a77/R/methods.R#L981-L983). Correspondingly, the GMPD(-ratio) is one, so multiplying it to the threshold does not change the threshold. Overall, I think the new `_diff` (and `_ratio`) variants of the `**pd` statistics are not needed within `plot.vsel()` (but I haven't tried it out without these yet).

Reply:
Marking this conversation as "resolved" because this problem has been circumvented (for this PR) by moving the new "mixed deltas" plot, the new column behavior of `summary.vsel()`, and some other changes to branch [`mixed_deltas_plot`](https://github.com/stan-dev/projpred/tree/mixed_deltas_plot). In the PR for that `mixed_deltas_plot` branch, this needs to be taken into account.